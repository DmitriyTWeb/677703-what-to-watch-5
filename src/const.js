export const RatingDesc = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const DEFAULT_RENDERED_FILMS_COUNT = 8;
export const PLAYING_TIMEOUT = 1000;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  MYLIST: `/mylist`,
  FILMS_ID: `/films/:id`,
  PLAYER_ID: `/player/:id`,
  REVIEW_ID: `/films/:id/review`,
  ERROR: `/error`,
};

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  PROMO: `/films/promo`,
};
