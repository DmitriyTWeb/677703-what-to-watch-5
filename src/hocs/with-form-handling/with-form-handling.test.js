import React from "react";
import renderer from "react-test-renderer";
import withFromHandling from "./with-form-handling";
import {Provider} from "react-redux";
import configureStore from 'redux-mock-store';
import {mockFilm} from "../../const.js";


const noop = () => {};

const MockComponent = () => {
  return (
    <div>
    </div>
  );
};

const mockStore = configureStore([]);
const MockComponentWrapped = withFromHandling(MockComponent);

describe(`Test withFromHandling with internal Component connected to Redux`, () => {
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore({});
    store.dispatch = jest.fn();

    tree = renderer.create(
        <Provider store={store}>
          <MockComponentWrapped
            postCommentAction={noop}
            redirectAction={noop}
            filmId={mockFilm.id}
          />
        </Provider>
    );
  });

  it(`withFromHandling is rendered correctly`, () => {
    expect(tree).toMatchSnapshot();
  });
});
