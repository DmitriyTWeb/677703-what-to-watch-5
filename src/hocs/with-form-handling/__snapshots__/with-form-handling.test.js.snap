// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test withFromHandling with internal Component connected to Redux withFromHandling is rendered correctly 1`] = `<div />`;
